From d27a29a155c44167961dd6829cb7b13657451485 Mon Sep 17 00:00:00 2001
From: Richard Zhu <hongxing.zhu@nxp.com>
Date: Tue, 7 Mar 2017 09:12:09 +0800
Subject: [PATCH 1551/5242] MLK-14325-3 PCI: designware: Remove the redundant
 codes

commit  68da752273537fb960dcc3a35d6bb3f7a7b45e00 from
https://source.codeaurora.org/external/imx/linux-imx.git

Remove the redundant codes in pcie-designware driver.
Because that the class configration had been
contained in 4.9 kernel.

Signed-off-by: Richard Zhu <hongxing.zhu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/pci/controller/dwc/pcie-designware-host.c |    5 -----
 1 file changed, 5 deletions(-)

diff --git a/drivers/pci/controller/dwc/pcie-designware-host.c b/drivers/pci/controller/dwc/pcie-designware-host.c
index fc2a7e9..4223a9c 100644
--- a/drivers/pci/controller/dwc/pcie-designware-host.c
+++ b/drivers/pci/controller/dwc/pcie-designware-host.c
@@ -714,11 +714,6 @@ void dw_pcie_setup_rc(struct pcie_port *pp)
 	val |= 0x00ff0100;
 	dw_pcie_writel_dbi(pci, PCI_PRIMARY_BUS, val);
 
-	/* program correct class for RC */
-	val = dw_pcie_readl_dbi(pci, PCI_CLASS_REVISION);
-	val |= PCI_CLASS_BRIDGE_PCI << 16;
-	dw_pcie_writel_dbi(pci, val, PCI_CLASS_REVISION);
-
 	/* Setup command register */
 	val = dw_pcie_readl_dbi(pci, PCI_COMMAND);
 	val &= 0xffff0000;
-- 
1.7.9.5

